src/xdp-app-info-flatpak.c

```diff
@@ -820,6 +848,10 @@ xdp_app_info_flatpak_new (int      pid,
   if (has_network)
     flags |= XDP_APP_INFO_FLAG_HAS_NETWORK;
 
+  flatpak_instance = flatpak_instance_from_metadata (metadata, error);
```

Comment:
FlatpakInstance does a lot of what XdpAppInfoFlatpak does as well. Might make sense to replace the logic in here with calls to FlatpakInstance, and move the rest of the stuff in here to FlatpakInstance.